tests/test_sig_stfl.c

```diff
@@ -379,22 +379,45 @@ static OQS_STATUS sig_stfl_test_secret_key(const char *method_name) {
 	OQS_STATUS rc = OQS_SUCCESS;
 	OQS_SIG_STFL_SECRET_KEY *sk = NULL;
 
+	OQS_SIG_STFL *sig_obj = NULL;
+	uint8_t *public_key = NULL;
+
+	printf("================================================================================\n");
+	printf("Create stateful Signature  %s\n", method_name);
+	printf("================================================================================\n");
+
+	sig_obj = OQS_SIG_STFL_new(method_name);
+	if (sig_obj == NULL) {
+		fprintf(stderr, "ERROR: OQS_SIG_STFL_new failed\n");
+		goto err;
+	}
+
+	public_key = malloc(sig_obj->length_public_key * sizeof(uint8_t));
+
+	printf("================================================================================\n");
+	printf("Create stateful Secret Key  %s\n", method_name);
+	printf("================================================================================\n");
+
 	sk = OQS_SIG_STFL_SECRET_KEY_new(method_name);
 	if (sk == NULL) {
 		fprintf(stderr, "ERROR: OQS_SECRET_KEY_new failed\n");
 		goto err;
 	}
 
 	printf("================================================================================\n");
-	printf("Create stateful Secret Key  %s\n", method_name);
+	printf("Generate keypair  %s\n", method_name);
 	printf("================================================================================\n");
+	rc = OQS_SIG_STFL_keypair(sig_obj, public_key, sk);
```

Comment:
Ah, you have to replace this function with `sig_stfl_KATs_keygen`, similar to line 308. 
Reason: at some parameter of XMSS, key generation takes about hours to finish. 
I think some variants of LMS also have very long key generation time. 
So that's why in function `sig_stfl_KATs_keygen`, it decides to generate keypair from the actual keypair function, or just simply load the key from KATs.

Reply:
I hope this will fix problems with XMSS. @ashman-p

Reply:
Added temp logic to exclude algs that take too long to generate keys or otherwise don't have a KAT file.